test/integration/scheduler/taint_test.go

```diff
@@ -571,3 +573,359 @@ func TestTaintNodeByCondition(t *testing.T) {
+		100,              // Eviction limiter QPS
+		100,              // Secondary eviction limiter QPS
+		50,               // Large cluster threshold
+		0.55,             // Unhealthy zone threshold
+		true,             // Run taint manager
+		true,             // Use taint based evictions
+		false,            // Enabled TaintNodeByCondition feature
+	)
+	if err != nil {
+		t.Errorf("Failed to create node controller: %v", err)
+		return
+	}
+
+	go nc.Run(context.stopCh)
+
+	// Waiting for all controller sync.
+	externalInformers.Start(context.stopCh)
+	externalInformers.WaitForCacheSync(context.stopCh)
+	informers.Start(context.stopCh)
+	informers.WaitForCacheSync(context.stopCh)
+
+	nodeRes := v1.ResourceList{
+		v1.ResourceCPU:    resource.MustParse("4000m"),
+		v1.ResourceMemory: resource.MustParse("16Gi"),
+		v1.ResourcePods:   resource.MustParse("110"),
+	}
+	node1Taints := []v1.Taint{
+		{
+			Key:    schedulerapi.TaintNodeUnreachable,
+			Effect: v1.TaintEffectNoExecute,
+		}}
+
+	node0 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "node-0",
+			Labels: map[string]string{v1.LabelZoneRegion: "region1", v1.LabelZoneFailureDomain: "zone1"},
+		},
+		Spec: v1.NodeSpec{},
+		Status: v1.NodeStatus{
+			Capacity:    nodeRes,
+			Allocatable: nodeRes,
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   v1.NodeReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+	node1 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "node-1",
+			Labels: map[string]string{v1.LabelZoneRegion: "region1", v1.LabelZoneFailureDomain: "zone1"},
+		},
+		Spec: v1.NodeSpec{},
+		Status: v1.NodeStatus{
+			Capacity:    nodeRes,
+			Allocatable: nodeRes,
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   v1.NodeReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+	node2 := &v1.Node{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   "node-2",
+			Labels: map[string]string{v1.LabelZoneRegion: "region1", v1.LabelZoneFailureDomain: "zone1"},
+		},
+		Spec: v1.NodeSpec{},
+		Status: v1.NodeStatus{
+			Capacity:    nodeRes,
+			Allocatable: nodeRes,
+			Conditions: []v1.NodeCondition{
+				{
+					Type:   v1.NodeReady,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+	}
+	if _, err := cs.CoreV1().Nodes().Create(node0); err != nil {
+		t.Errorf("Failed to create node, err: %v", err)
+	}
+	if _, err := cs.CoreV1().Nodes().Create(node1); err != nil {
+		t.Errorf("Failed to create node, err: %v", err)
+	}
+	if _, err := cs.CoreV1().Nodes().Create(node2); err != nil {
+		t.Errorf("Failed to create node, err: %v", err)
+	}
+
+	node1.Status.Conditions = []v1.NodeCondition{
+		{
+			Type:   v1.NodeReady,
+			Status: v1.ConditionUnknown,
+		},
+	}
+
+	// Update node condition.
+	err = updateNodeStatus(cs, node1)
+	if err != nil {
+		t.Fatalf("Cannot update node: %v", err)
+	}
+
+	if err := waitForNodeTaints(cs, node1, node1Taints); err != nil {
+		t.Errorf("Failed to taint node <%s>, err: %v", node1.Name, err)
+	}
+
```

Comment:
We should also implement the function included in original e2e tests:

- Default pods are applied tolerationSeconds=300, and evicted after 300 seconds
- Pod with tolerationSeconds=60 is evicted after 60 seconds
- Pods having tolerations matching unreachable are kept

And also remove the `Unknow` status to see if the taint is taken off from the node.